codes/ruby/utils/list_node.rb

```diff
@@ -0,0 +1,51 @@
+=begin
+File: list_node.rb
+Created Time: 2024-03-18
+Author: Xuan Khoa Tu Nguyen (ngxktuzkai2000@gmail.com)
+=end
+
+### Definition for a singly-linked list node ###
+class ListNode
+  attr_accessor :val  # 节点值
+  attr_accessor :next # 指向下一节点的引用
+
+  def initialize(val=nil, next_node=nil)
+    @val = val || 0
+    @next = next_node
+  end
+end
+
+class LinkedList
+  class << self
+    # Generate a linked list from an array
```

Comment:
Please refer to Python version to use Chinese comments.

(I will update the comments of Java version to Chinese too)